--- /usr/share/pve-manager/js/pvemanagerlib.js.orig	2018-06-04 11:21:26.419121310 +0800
+++ /usr/share/pve-manager/js/pvemanagerlib.js	2018-06-04 11:52:58.628592707 +0800
@@ -5303,6 +5303,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
 	['comstar', 'Comstar'],
+	['freenas', 'FreeNAS-API'],
 	[ 'istgt', 'istgt'],
 	[ 'iet', 'IET']
     ]
@@ -27145,6 +27146,13 @@
 		value: '',
 		fieldLabel: gettext('Target group'),
 		allowBlank: true
+	    },
+	    {
+		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		name: 'freenas_user',
+		value: '',
+		fieldLabel: gettext('FreeNAS User'),
+		allowBlank: true
 	    }
 	];
 
@@ -27176,6 +27184,13 @@
 		value: '',
 		fieldLabel: gettext('Host group'),
 		allowBlank: true
+	    },
+	    {
+		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		name: 'freenas_password',
+		value: '',
+		fieldLabel: gettext('FreeNAS Password'),
+		allowBlank: true
 	    }
 	];
 
