--- pvemanagerlib-5.2-9.js	2018-10-16 07:17:21.915593478 -0400
+++ pvemanagerlib.js	2018-10-21 00:46:12.524172694 -0400
@@ -5597,6 +5597,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
 	['comstar', 'Comstar'],
+	['freenas', 'FreeNAS-API'],
 	[ 'istgt', 'istgt'],
 	[ 'iet', 'IET'],
 	[ 'LIO', 'LIO']
@@ -28574,6 +28575,7 @@
 	data: {
 	    isLIO: false,
 	    isComstar: true,
+	    isFreeNAS: false,
 	    hasWriteCacheOption: true
 	}
     },
@@ -28586,10 +28588,18 @@
 	    }
 	},
 	changeISCSIProvider: function(f, newVal, oldVal) {
+	    var me = this;
 	    var vm = this.getViewModel();
 	    vm.set('isLIO', newVal === 'LIO');
 	    vm.set('isComstar', newVal === 'comstar');
-	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
+	    vm.set('isFreeNAS', newVal === 'freenas');
+	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'freenas' || newVal === 'istgt');
+	    if (newVal !== 'freenas') {
+		me.lookupReference('freenas_use_ssl_field').setValue(false);
+		me.lookupReference('freenas_apiv4_host_field').setValue('');
+		me.lookupReference('freenas_user_field').setValue('');
+		me.lookupReference('freenas_password_field').setValue('');
+	    }
 	}
     },
 
@@ -28648,8 +28658,28 @@
 		name: 'comstar_tg',
 		value: '',
 		fieldLabel: gettext('Target group'),
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: me.isCreate ? { hidden: '{!isComstar}' } : { hidden: '{!isComstar}' },
 		allowBlank: true
+	    },
+	    {
+		xtype: 'proxmoxcheckbox',
+		name: 'freenas_use_ssl',
+		reference: 'freenas_use_ssl_field',
+		inputId: 'freenas_use_ssl_field',
+		checked: false,
+		bind: me.isCreate ? { hidden: '{!isFreeNAS}' } : { hidden: '{!isFreeNAS}' },
+		uncheckedValue: 0,
+		fieldLabel: gettext('API use SSL')
+	    },
+	    {
+		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		name: 'freenas_user',
+		reference: 'freenas_user_field',
+		inputId: 'freenas_user_field',
+		value: '',
+		fieldLabel: gettext('API Username'),
+		bind: me.isCreate ? { hidden: '{!isFreeNAS}' } : { hidden: '{!isFreeNAS}' },
+		allowBlank: me.isFreeNAS ? false : true
 	    }
 	];
 
@@ -28680,7 +28710,7 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'comstar_hg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: me.isCreate ? { hidden: '{!isComstar}' } : { hidden: '{!isComstar}' },
 		fieldLabel: gettext('Host group'),
 		allowBlank: true
 	    },
@@ -28688,9 +28718,27 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'lio_tpg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isLIO}' } : { hidden: '{!isLIO}' },
-		allowBlank: false,
-		fieldLabel: gettext('Target portal group')
+		bind: me.isCreate ? { hidden: '{!isLIO}' } : { hidden: '{!isLIO}' },
+		fieldLabel: gettext('Target portal group'),
+		allowBlank: true
+	    },
+	    {
+		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		name: 'freenas_apiv4_host',
+		reference: 'freenas_apiv4_host_field',
+		value: '',
+		bind: me.isCreate ? { hidden: '{!isFreeNAS}' } : { hidden: '{!isFreeNAS}' },
+		fieldLabel: gettext('API IPv4 Host'),
+		allowBlank: true
+	    },
+	    {
+		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		name: 'freenas_password',
+		reference: 'freenas_password_field',
+		value: '',
+		bind: me.isCreate ? { hidden: '{!isFreeNAS}' } : { hidden: '{!isFreeNAS}' },
+		fieldLabel: gettext('API Password'),
+		allowBlank: true
 	    }
 	];
 
