--- apidoc.js.orig	2021-11-15 10:07:34.000000000 -0500
+++ apidoc.js	2021-12-06 08:04:01.648822707 -0500
@@ -44064,6 +44064,31 @@
                            "type" : "string",
                            "typetext" : "<string>"
                         },
+                        "freenas_user" : {
+                           "description" : "FreeNAS user for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "freenas_password" : {
+                           "description" : "FreeNAS password for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "freenas_use_ssl" : {
+                           "description" : "FreeNAS API access via SSL",
+                           "optional" : 1,
+                           "type" : "boolean",
+                           "typetext" : "<boolean>"
+                        },
+                        "freenas_apiv4_host" : {
+                           "description" : "FreeNAS API Host via IPv4",
+                           "format" : "address",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "fuse" : {
                            "description" : "Mount CephFS through FUSE.",
                            "optional" : 1,
@@ -44275,6 +44300,12 @@
                            "type" : "boolean",
                            "typetext" : "<boolean>"
                         },
+                        "target" : {
+                           "description" : "iSCSI target.",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "transport" : {
                            "description" : "Gluster transport: tcp or rdma",
                            "enum" : [
@@ -44547,6 +44578,31 @@
                      "optional" : 1,
                      "type" : "string",
                      "typetext" : "<string>"
+                  },
+                  "freenas_user" : {
+                     "description" : "FreeNAS user for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "freenas_password" : {
+                     "description" : "FreeNAS password for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "freenas_use_ssl" : {
+                     "description" : "FreeNAS API access via SSL",
+                     "optional" : 1,
+                     "type" : "boolean",
+                     "typetext" : "<boolean>"
+                  },
+                  "freenas_apiv4_host" : {
+                     "description" : "FreeNAS API Host via IPv4",
+                     "format" : "address",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
                   },
                   "fuse" : {
                      "description" : "Mount CephFS through FUSE.",
