--- apidoc.js.orig	2019-07-15 15:45:00.000000000 -0400
+++ apidoc.js	2019-08-07 13:28:29.292382229 -0400
@@ -35001,6 +35001,31 @@
                            "type" : "string",
                            "typetext" : "<string>"
                         },
+                        "freenas_user" : {
+                           "description" : "FreeNAS user for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "freenas_password" : {
+                           "description" : "FreeNAS password for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "freenas_use_ssl" : {
+                           "description" : "FreeNAS API access via SSL",
+                           "optional" : 1,
+                           "type" : "boolean",
+                           "typetext" : "<boolean>"
+                        },
+                        "freenas_apiv4_host" : {
+                           "description" : "FreeNAS API Host via IPv4",
+                           "format" : "address",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "fuse" : {
                            "description" : "Mount CephFS through FUSE.",
                            "optional" : 1,
@@ -35157,6 +35182,12 @@
                            "type" : "boolean",
                            "typetext" : "<boolean>"
                         },
+                        "target" : {
+                           "description" : "iSCSI target.",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "transport" : {
                            "description" : "Gluster transport: tcp or rdma",
                            "enum" : [
@@ -35362,6 +35393,31 @@
                      "optional" : 1,
                      "type" : "string",
                      "typetext" : "<string>"
+                  },
+                  "freenas_user" : {
+                     "description" : "FreeNAS user for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "freenas_password" : {
+                     "description" : "FreeNAS password for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "freenas_use_ssl" : {
+                     "description" : "FreeNAS API access via SSL",
+                     "optional" : 1,
+                     "type" : "boolean",
+                     "typetext" : "<boolean>"
+                  },
+                  "freenas_apiv4_host" : {
+                     "description" : "FreeNAS API Host via IPv4",
+                     "format" : "address",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
                   },
                   "fuse" : {
                      "description" : "Mount CephFS through FUSE.",
